rfcs/001 - EC2 schema.md

```diff
@@ -0,0 +1,113 @@
+# 001 - EC2 Schema
+
+## Current Status
+
+### Proposed
+
+2022-07-18
+
+### Accepted
+
+YYYY-MM-DD
+
+#### Approvers
+
+- LuisFer De Pombo <luisfer@iasql.com>
+- David Ellis <david@alantechnologies.com>
+
+### Implementation
+
+- [ ] Implemented: [One or more PRs](https://github.com/iasql/iasql-engine/some-pr-link-here) YYYY-MM-DD
+- [ ] Revoked/Superceded by: [RFC ###](./000 - RFC Template.md) YYYY-MM-DD
+
+## Author(s)
+
+- Alejandro Guillen <alejandro@iasql.com>
+
+## Summary
+
+Adding new features to get closer to the goal of the EC2 module's completeness adds more complexity to the model. This complexity directly affects the user experience, especially for new users, since they can feel overwhelmed if they encounter a complicated model.
+
+The definition of a solid base now will imply that new features added to this module will not translate into breaking changes to the model.
+
+Taking a look at the AWS EC2 console options, we would still need to add the following to reach EC2 module completeness:
+
+- Instance type - Availability zone relationship
+- Launch templates
+- Spot instances
+- Dedicated hosts
+- Saving plans
+- Reserved instances
+- Scheduled instances
+- Capacity reservations
+- Auto Scaling Groups
+- AMIs management
+- EBS volume snapshots and life cycle
+- Elastic IP integration
+- Placement groups
+- Network interfaces
+
+Almost all the features listed will need a direct integration with the `instance` table, needing to add at least eight more columns to the model. Those columns are only for integration, but the `instance` model still needs more properties to be defined to be complete. Some of those missing properties are: instance auto-recovery, shutdown behaviour, stop - hibernate behaviour, termination protection, stop protection, detailed cloudWatch monitoring, elastic GPU, credit specification, tenancy, RAM disk ID, kernel ID or Metadata accessible (I might be missing others).
```

Comment:
Wouldn’t some, or most, of the properties described at the end be part of the `instance_metadata` table instead of the `instance` table?

Reply:
I don't think so since all of them are possible inputs for instance creation